### PR TITLE
Rs 19 annotations favourites handler

### LINKED ISSUE #19: Add AnnotationsFavouritesHandler handler
The current implementation to handle favourites creates Link-objects into the homefolder and manages the security through the placefulworkflow.

The Favourites-Handler is an adapter providing the IFavouritesHandler interface (https://github.com/4teamwork/ftw.dashboard.portlets.favourites/blob/master/ftw/dashboard/portlets/favourites/configure.zcml#L17)

The implementation of the DefaultFavouritesHandler has some disadvantages if you have multilingual sites. For this reason, we have to implement a new adapter which stores the favourites into the annotations.

Todos:

- [x] add interface-test (example: https://github.com/4teamwork/ftw.activity/blob/master/ftw/activity/tests/test_filters.py#L26)
- [x] interface-adjustments: https://github.com/4teamwork/ftw.dashboard.portlets.favourites/blob/master/ftw/dashboard/portlets/favourites/interfaces.py#L4 (The interface is too specific. We have to do it more general).
  - [x] rename `create_favourites_folder` to `create_favourites_container`
  - [x] rename `get_favourites_folder` to `get_favourites_container`
  - [x] remove `get_favourites_filter_query`
  - [x] remove `get_favourite_folder_name`
  - [x] remove `get_home_folder``
- [x] register a new generic-setup profile: `annotationstorage`
- [x] create a new browserlayer for the new profile: `IFavouritesAnnotationStorageLayer` (example: https://github.com/4teamwork/usb.guidelines/blob/adaf14e0fa0264d9aedab8b5b4054e5f034b6ade/usb/guidelines/profiles/default/browserlayer.xml)
- [x] register a new adapter for the new browserlayer: `AnnotationStorgaeFavouritesHandler` (example: https://github.com/plone/plone.app.contentmenu/blob/master/plone/app/contentmenu/configure.zcml#L98)
- [x] register a new testlayer `FavouritesAnnotationStorageLayer` which also installs the news`annotationstorage` profile
- [x] Refactor the `get_favourites`-method of the `DefaultFavouritesHandler` (https://github.com/4teamwork/ftw.dashboard.portlets.favourites/blob/master/ftw/dashboard/portlets/favourites/adapter.py#L103). The current implementation returns brains. It should return the attributes used in the template in an ordered list with dicts (as the implementation of the `AnnotationStorgaeFavouritesHandler`
- [x] Adjust the portlet-template that it can handle the new return-value of the `get_favourites` method (https://github.com/4teamwork/ftw.dashboard.portlets.favourites/blob/675c907b59aeb3e4f1e17de309c4e6e1dfe33363/ftw/dashboard/portlets/favourites/portlets/templates/favourites.pt#L16)
- [x] implement the necessary functions for the new adapter
  - [x] see https://github.com/4teamwork/ftw.labels/blob/88d98421a26d1b31fbcbdc103768da98bcd7c514/ftw/labels/labeling.py#L51 as an example how to use annotations
  - [x] use the plone root as the annotation-context
  - [x] use `OOBTree`, `PersistentList` and `PersistentDict` for the structure
  - [x] The `get_favourites`-method in the `FavouritesAnnotationStorageLayer` adapter should return an ordered list with dicts with the given favourites
  - [x] The structure should be like this:
~~PersistentDict(~~
  ~~userid1=PersistentDict(id="", title="", remote_url="", sort_order=""),~~
  ~~userid2=PersistentDict(id="", title="", remote_url="", sort_order=""))~~
=> see answer of @jone:
```python
from BTrees.OOBTree import OOBTree
from persistent.list import PersistentList
from persistent.mapping import PersistentMapping

OOBTree(
    user1=PersistentList([PersistentMapping(id="", title="", remote_url="", sort_order=""), PersistentMapping(..)]),
    user2=PersistentList([PersistentMapping(...),]),
```
- [x] Tests

### PR DESCRIPTION
**Pull request by raphael-s**
Refactors the favourites portlet to use annotations instead of a user home-folder.

If a user adds a favourite, the item will be added to a persistendDict on the plone root annotations.

closes #19

### CODE REVIEW

**Comment by Tschanzt** [comment]
@raphael-s the action to add a favorite still has the requirment of a userfolder existing. this isn't a requirment anymore with your implementation so you might want to change that. 

Also i would rephrase the profile title to "annotationstorage support for ftw.dashboard.portlets.favourites" since it doesn't install the default profile.

**Comment by raphael-s** [comment]
@Tschanzt those should both be fine now.
